flexmeasures/data/models/planning/storage.py

```diff
@@ -166,13 +169,209 @@ def schedule(
+                self.flex_model["soc_at_start"] = self.sensor.get_attribute(
+                    "soc_in_mwh"
+                )
+            else:
+                self.flex_model["soc_at_start"] = 0
+
+        # Check for round-trip efficiency
+        if (
+            "roundtrip_efficiency" not in self.flex_model
+            or self.flex_model["roundtrip_efficiency"] is None
+        ):
+            # Get default from sensor, or use 100% otherwise
+            self.flex_model["roundtrip_efficiency"] = self.sensor.get_attribute(
+                "roundtrip_efficiency", 1
+            )
+        if (
+            self.flex_model["roundtrip_efficiency"] <= 0
+            or self.flex_model["roundtrip_efficiency"] > 1
+        ):
+            raise ValueError("roundtrip_efficiency expected within the interval (0, 1]")
+
+        self.ensure_soc_min_max()
+
+        # Now it's time to check if our flex configurations hold up to basic expectations
+        self.flex_model = self.flex_model_schema().load(self.flex_model)
+        self.flex_context = FlexContextSchema().load(self.flex_context)
+
+        self.check_soc_min_max_and_targets()
+
+        # Make SOC targets into a series for easier use
+        self.flex_model["soc_targets"] = build_soc_targets(
+            self.flex_model.get("soc_targets", []),
+            self.start,
+            self.end,
+            self.sensor.event_resolution,
+        )
+
+        return self.flex_model
+
+    def get_min_max_targets(self) -> tuple(float | None):
+        min_target = None
+        max_target = None
+        if "soc_targets" in self.flex_model and len(self.flex_model["soc_targets"]) > 0:
+            min_target = min(
+                [target["value"] for target in self.flex_model["soc_targets"]]
+            )
+            max_target = max(
+                [target["value"] for target in self.flex_model["soc_targets"]]
+            )
+        return min_target, max_target
+
+    def get_min_max_soc_on_sensor(
+        self, adjust_unit: bool = False
+    ) -> tuple(float | None):
+        soc_min_sensor = self.sensor.get_attribute("min_soc_in_mwh", None)
+        soc_max_sensor = self.sensor.get_attribute("max_soc_in_mwh", None)
+        if adjust_unit:
+            if soc_min_sensor and self.flex_model.get("soc_unit") == "kWh":
+                soc_min_sensor *= 1000  # later steps assume soc data is kWh
+            if soc_max_sensor and self.flex_model.get("soc_unit") == "kWh":
+                soc_max_sensor *= 1000
+        return soc_min_sensor, soc_max_sensor
+
+    def check_soc_min_max_and_targets(self):
+        """
+        Check if targets or min and max values are out of any existing known bounds
+        """
+        min_target, max_target = self.get_min_max_targets()
+        soc_min_sensor, soc_max_sensor = self.get_min_max_soc_on_sensor()
+
+        if min_target and min_target < 0:
+            raise ValueError(f"Lowest SOC target {min_target} MWh lies below 0.")
```

Comment:
We should not make our own assumptions about the feasibility of targets. The next if-statement already checks its feasibility against a sensor attribute. Also see my other comment about the interpretation of that sensor attribute as a default rather than as a constraint.